### PR TITLE
Apache derby db setup

### PR DESCRIPTION
**Pull request by natcrossman**
Here is the list of changes that would get integrated into the project, if this pull request gets accepted:
1.  Part one of Issue #274 is completed
a. Apache Derby's embedded database is configured with our Gradle build system
2. Add's a new JUnit test to our test library
a. Creates an embedded memory database and perform certain operations to make sure Derby is working properly
3. Adds necessary images to our wiki pages
4. Fixes copious amounts of CheckStyle violations

### CODE REVIEW

File: src/edu/wright/cs/raiderplanner/controller/MenuController.java [hunk 1]
```diff
@@ -148,7 +148,7 @@
 	private DropShadow moduleHoverShadow = new DropShadow(screenAverage * 0.02, 0, 0, Color.BLACK);
 	private InnerShadow modulePressedShadow = new InnerShadow(screenAverage * 0.017, 0, 0,
 			Color.BLACK);
-
+	private Stage  stage = null;
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/controller/MenuController.java]: Please remove the extra space between `Stage` and `stage`.

File: src/edu/wright/cs/raiderplanner/model/Exam.java [hunk 2]
```diff
@@ -45,26 +45,51 @@ protected void replace(VersionControlEntity receivedVce) {
 		super.replace(receivedVce);
 	}
 
-	// public methods
-
-	// getters
+	/**
+	 * Get's the Resit value.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Exam.java]: `Get's` should be `Gets`. Please remove the apostrophe, as well as the other like it in this file.

File: test/edu/wright/cs/raiderplanner/model/ApacheDerbyDatabaseTest.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2017 - Nathaniel Crossman
```
Comment (rsanchez-wsu) [inline test/edu/wright/cs/raiderplanner/model/ApacheDerbyDatabaseTest.java]: Year should be 2018.

File: src/edu/wright/cs/raiderplanner/controller/MenuController.java (hunk 1 above)
Comment (natcrossman) [reply src/edu/wright/cs/raiderplanner/controller/MenuController.java]: How did you even see that extra white space!

File: test/edu/wright/cs/raiderplanner/model/ApacheDerbyDatabaseTest.java (hunk 3 above)
Comment (natcrossman) [reply test/edu/wright/cs/raiderplanner/model/ApacheDerbyDatabaseTest.java]: Not sure why Eclipse generated the wrong year.

**Comment by natcrossman** [comment]
Ok I fixed them

File: src/edu/wright/cs/raiderplanner/model/Exam.java [hunk 4]
```diff
@@ -45,26 +45,51 @@ protected void replace(VersionControlEntity receivedVce) {
...
+	/**
+	 * This Getter method gets the Resit value.
+	 * @return
+	 */
 	public Exam getResit() {
 		return resit;
 	}
 
+	/**
+	 * Get's the Exam event time slot.
```
Comment (rsanchez-wsu) [inline src/edu/wright/cs/raiderplanner/model/Exam.java]: This should have been changed when you fixes the other occurrence of `Get's` in this file.

**Comment by natcrossman** [comment]
Sorry! I did not see that one.

**Pull request merged**